### PR TITLE
fix scanning QRs and adding wallets issue

### PR DESCRIPTION
**Pull request by benisgold**
Fixes RNBW-2468, RNBW-2628

## What changed (plus any additional context for devs)
Check that the function `contextValue?.setIsSearchModeEnabled` exists before invoking it. This function doesn't exist if the wallet doesn't have an ens, which is what was causing the bug.

## PoW (screenshots / screen recordings)
https://user-images.githubusercontent.com/15272675/155228346-6ca2a46b-c972-49e2-b96a-6822272a27f9.mp4

## Dev checklist for QA: what to test
Make sure that wallets that don't have ens names can be watched.

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels?

### CODE REVIEW

**Review by skylarbarrera — APPROVED** [review]
🙏

File: src/components/showcase/ShowcaseHeader.js [hunk 1]
```diff
@@ -153,7 +153,9 @@ export function Header() {
   const { handleSetSeedPhrase, handlePressImportButton } = useImportingWallet();
 
   const onWatchAddress = useCallback(() => {
-    contextValue?.setIsSearchModeEnabled(false);
+    if (contextValue?.setIsSearchModeEnabled) {
+      contextValue?.setIsSearchModeEnabled(false);
```
Comment (brunobar79) [inline src/components/showcase/ShowcaseHeader.js:157]: You don't need the optional here since you're already checking on the line above

**Review by brunobar79 — APPROVED** [review]
🙏

**Comment by osdnk** [comment]
@benisgold 
I think this should also fix https://linear.app/rainbow/issue/RNBW-2628/crash-after-pressing-watch-this-wallet-after-navigating-to-the-wallet, but can you verify this, please?

**Pull request merged**